src/lib/model/lib/shared/index.js

```diff
@@ -275,6 +277,171 @@ const mojaloopTransactionRequestToInternal = (external) => {
+            initiator: quote.transactionType.initiator,
+            initiatorType: quote.transactionType.initiatorType
+        };
+
+        if (quote.fees) {
+            internal.feesAmount = quote.fees.amount;
+            internal.feesCurrency = quote.fees.currency;
+        }
+
+        if (quote.geoCode) {
+            internal.geoCode = quote.geoCode;
+        }
+
+        if (quote.note) {
+            internal.note = quote.note;
+        }
+
+        return internalQuote;
+    });
+
+    internal.individualQuotes = internalIndividualQuotes;
+
+    return internal;
+};
+
+/**
+ * Converts an internal bulk quotes response to mojaloop form
+ *
+ * @returns {object}
+ */
+const internalBulkQuotesResponseToMojaloop = (internal) => {
+    const individualQuoteResults = internal.individualQuotes.map((quote) => {
+        const externalQuote = {
+            quoteId: quote.quoteId,
+            transferAmount: {
+                amount: quote.transferAmount,
+                currency: quote.transferAmountCurrency,
+            },
+            ilpPacket: quote.ilpPacket,
+            condition: quote.ilpCondition
+        };
+
+        if (quote.payeeReceiveAmount) {
+            externalQuote.payeeReceiveAmount = {
+                amount: quote.payeeReceiveAmount,
+                currency: quote.payeeReceiveAmountCurrency
+            };
+        }
+
+        if (quote.payeeFspFeeAmount) {
+            externalQuote.payeeFspFee = {
+                amount: quote.payeeFspFeeAmount,
+                currency: quote.payeeFspFeeAmountCurrency
+            };
+        }
+
+        if (quote.payeeFspCommissionAmount) {
+            externalQuote.payeeFspCommission = {
+                amount: quote.payeeFspCommissionAmount,
+                currency: quote.payeeFspCommissionAmountCurrency
+            };
+        }
+
+        return externalQuote;
+    });
+    const external = {
+        individualQuoteResults,
+        expiration: internal.expiration,
+    };
+
+    if (internal.geoCode) {
+        external.geoCode = internal.geoCode;
+    }
+
+    if (internal.extensionList) {
+        external.extensionList = internal.extensionList;
+    }
+
+    return external;
+};
+
+/**
+ * Converts a mojaloop bulk transfer prepare request to internal form
+ *
+ * @returns {object}
+ */
+const mojaloopBulkPrepareToInternalBulkTransfer = (external, bulkQuotes, ilpSecret) => {
+    let internal = null;
+    if (bulkQuotes) {
+        // create a map of internal individual quotes payees indexed by quotedId, for faster lookup
+        const internalQuotesPayeesByQuoteId = {};
+        
+        for (const quote of bulkQuotes.internalRequest.individualQuotes) {
+            internalQuotesPayeesByQuoteId[quote.quoteId] = quote.to;
+        }
+        
+        // create a map of external individual transfers indexed by quotedId, for faster lookup
+        const externalTransferIdsByQuoteId = {};
+        const ilp = new Ilp({ secret: ilpSecret });
```

Comment:
ilpSecret never changes during the lifetime of the scheme adapter process. It is more efficient to Instantiate the ilp utility class once only, not every time a function is called. It would be better to not pass the ilpSecret to this function, instead pass an instance of the ilp class managed at the next layer up, or better still, pass the `getTransactionObject` function from the upper layer instance itself.